src/utils.jl

```diff
@@ -370,42 +377,44 @@ function _transfer_solution!(data,pv,fh,ip_geo,ip_field,cellset_,val_buffer,val,
     grid = dh.grid
     # actual data
     local_dof_range = Ferrite.dof_range(fh, field_name)
-    _processreturn = length(process(val_buffer))
-
-    current_vertex_index = 1
+    _processreturndim = length(process(val_buffer))
 
     cell_geo_ref = Ferrite.getcells(grid, cellset_[1])
 
-    Ferrite.reinit!(pv, Ferrite.getcoordinates(grid,cellset_[1]), Tensors.Vec{dim}(ref_coords[current_vertex_index,:]))
+    Ferrite.reinit!(pv, Ferrite.getcoordinates(grid,cellset_[1]), Tensors.Vec{dim}(ref_coords[1,:]))
```

Comment:
if the first index is constant, can we resort to a vector instead of a matrix?

Reply:
I think I do not understand the question. You mean that we should repalce the matrix with a vector of vectors?

Reply:
ah nvm its just the initialization